### ISSUE TITLE
Publication range feature on publishers page not filtering as described

### ISSUE THREAD

**Issue by agmckee** [internetarchive/openlibrary#5955]
On a publisher page e.g. https://openlibrary.org/publishers/MIT_Press clicking or dragging a range across the graph does not result in the works being filtered as described.

I searched the issues but couldn't find anything that looked like a report of this issue.

### Evidence / Screenshot (if possible)
![openlibrary-filter-date-range](https://user-images.githubusercontent.com/722096/145307963-9aaee792-4964-402a-88f6-8c286bd575b7.png)

Tested in Firefox and Chromium, doesn't work in either.

### Relevant url?
[https://openlibrary.org/publishers/MIT_Press](https://openlibrary.org/publishers/MIT_Press)

### Steps to Reproduce
1. Go to the page of any publisher
2. Click a year or drag a range across years

* Actual: The works are not filtered down to that year or range
* Expected: The works are filtered to that year or range

### Details

- **Logged in (Y/N)?** No
- **Browser type/version?** Firefox 78.15 ESR, Chromium 90.0.4430.212
- **Operating system?** Debian GNU/Linux
- **Environment (prod/dev/local)?** prod

**Comment by agmckee** [comment]
Here's how it used to work: h
ttps://web.archive.org/web/20120510154805/http://openlibrary.org/publishers/A_Book_Apart

**Comment by agmckee** [comment]
I found that the code that handled the `plotselected` and `plotclick` events was removed seemingly when the code was moved into the graph module from its original location in the views. However the original functions won't work with the current carousel functions.

I was able to restore the functionality in my development branch by adding new `plotselected` and `plotclick` event handlers that handle the UI update while also emitting a custom event for the filter event with the date range data as a payload. I added a new event listener in Carousel that reacts to that event and updates the `searchParams` to include the `published_in` date filter and handle the UI update on refreshing the new, filtered works list.
